### ISSUE TITLE
ValueError('not enough values to unpack (expected 3, got 1)')

### ISSUE THREAD

**Issue by jonathan-anyword** [celery/kombu#2023]
Hi,
Recently I started getting the error.
I'm not familiar with kombu's data structure and I'm not sure how we getting into this situation.
Can someone direct me what I can check or test?

kombu/transport/virtual/exchange.py in <setcomp> at line 67
```
  def lookup(self, table, exchange, routing_key, default):
        return {
            queue for rkey, _, queue in table
            if rkey == routing_key
        }
```
Vars values:
```
default | None
exchange | "reply.celery.pidbox"
routing_key | "766541f1-fa51-3699-9303-50e4fa087f6e"
self | <kombu.transport.virtual.exchange.DirectExchange object at 0x7fa00c4b27f0>
table | [["766541f1-fa51-3699-9303-50e4fa087f6e\x06\x16\x06\x16766541f1-fa51-3699-9303-50e4fa087f6e.reply.celery.pidbox"]]
```

`kombu 5.3.7`
`celery 5.4.0`

Thanks

**Comment by thedrow** [comment]
Can you provide a traceback and a minimally reproducible test case?
Right now there's not enough information to reproduce this bug.

**Comment by yanivps** [comment]
I had the same error and found the solution:
What happens is that you have probably to celery workers connected to the same Redis backend and they do some syncing between them. In one of them you changed the `sep` value, i.e.:

```
"broker_transport_options": {
            "sep": ":",
            "queue_order_strategy": "priority",
        }
```

and in the other worker you did not which uses the default sep which is `x06x16`

Then when one is syncing the other they do not know how to split the queue name string because of the different separator.

Bottom line, keep the same value for `sep` in all of your workers

**Comment by VisualGadget** [comment]
Also `inspect` doesn't support custom separator =(

**Comment by SobhanKiani** [comment]
Hi, 
Can someone explain please that how can I define the `sep` value for different workers? 
The problem is that I have defined a default worker and one other worker. while the default worker works fine, the second worker raises the exception 
```ValueError('not enough values to unpack (expected 3, got 1)') ``` as described in this issue before.
So I think that I should set the same `sep` for the second worker but I don't know how. Neither Kombu nor Celery mentions this in their documentation, or at least I couldn't find it there.
The other problem is that when I run the same workers with the same configs locally, there is no problem most of the times and, I can't reproduce the same error (But sometimes randomly this problem happens). However in production this happens more often.
Thanks
